### PR TITLE
feat!: make `loading` and `error` properties readonly in the `useCart` composable

### PR DESCRIPTION
**Pull request by roziscoding**
## Description
- Change `useCart` composable and following API methods to **function declarations**:
  - `addProductsToCart`
  - `addConfigurableProductsToCart`
  - `addDownloadableProductsToCart`
  - `addVirtualProductsToCart`
  - `applyCouponToCart`
  - `removeCouponFromCart`
  - `cart`
  - `customerCart`
  - `removeItemFromCart`
  - `updateCartItems`

- Add JSDoc comments to `useCart` composable, its interfaces, and the related API methods.

## Related Issue

https://vsf.atlassian.net/browse/M2-375

## Motivation and Context

Documents the `useCart` with examples for the community.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/api-client/src/api/addVirtualProductsToCart/index.ts [hunk 1]
```diff
@@ -1,23 +1,29 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import addVirtualProductsToCart from './addVirtualProductsToCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/addVirtualProductsToCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/addConfigurableProductsToCart/index.ts [hunk 2]
```diff
@@ -1,23 +1,27 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import addConfigurableProductsToCart from './addConfigurableProductsToCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/addConfigurableProductsToCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/addDownloadableProductsToCart/index.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import addDownloadableProductsToCart from './addDownloadableProductsToCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/addDownloadableProductsToCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/applyCouponToCart/index.ts [hunk 4]
```diff
@@ -1,24 +1,30 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import applyCouponToCart from './applyCouponToCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/applyCouponToCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/cart/index.ts [hunk 5]
```diff
@@ -1,19 +1,25 @@
 import { ApolloQueryResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import { CartQuery, CartQueryVariables } from '../../types/GraphQL';
-import cart from './cart';
 import { Context } from '../../types/context';
+import { CartQuery, CartQueryVariables } from '../../types/GraphQL';
```
Comment (Frodigo) [inline packages/api-client/src/api/cart/index.ts:4]:
```suggestion
import type { CartQuery, CartQueryVariables } from '../../types/GraphQL';
```

File: packages/api-client/src/api/customerCart/index.ts [hunk 6]
```diff
@@ -1,23 +1,28 @@
 import { ApolloQueryResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
```
Comment (Frodigo) [inline packages/api-client/src/api/customerCart/index.ts:2]:
```suggestion
import type { CustomQuery } from '@vue-storefront/core';
```

File: packages/api-client/src/api/customerCart/index.ts [hunk 7]
```diff
@@ -1,23 +1,28 @@
...
-import { CustomerCartQuery } from '../../types/GraphQL';
-import customerCart from './customerCart';
 import { Context } from '../../types/context';
+import { CustomerCartQuery } from '../../types/GraphQL';
```
Comment (Frodigo) [inline packages/api-client/src/api/customerCart/index.ts:4]:
```suggestion
import type { CustomerCartQuery } from '../../types/GraphQL';
```

File: packages/api-client/src/api/customerCart/index.ts [hunk 8]
```diff
@@ -1,23 +1,28 @@
 import { ApolloQueryResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import { CustomerCartQuery } from '../../types/GraphQL';
-import customerCart from './customerCart';
 import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/customerCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/removeCouponFromCart/index.ts [hunk 9]
```diff
@@ -1,23 +1,29 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
-import removeCouponFromCart from './removeCouponFromCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/removeCouponFromCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/removeItemFromCart/index.ts [hunk 10]
```diff
@@ -1,23 +1,29 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
```
Comment (Frodigo) [inline packages/api-client/src/api/removeItemFromCart/index.ts:2]:
```suggestion
import type { CustomQuery } from '@vue-storefront/core';
```

File: packages/api-client/src/api/removeItemFromCart/index.ts [hunk 11]
```diff
@@ -1,23 +1,29 @@
...
-import removeItemFromCart from './removeItemFromCart';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/removeItemFromCart/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

File: packages/api-client/src/api/updateCartItems/index.ts [hunk 12]
```diff
@@ -1,23 +1,29 @@
 import { FetchResult } from '@apollo/client/core';
 import { CustomQuery } from '@vue-storefront/core';
```
Comment (Frodigo) [inline packages/api-client/src/api/updateCartItems/index.ts:2]:
```suggestion
import type { CustomQuery } from '@vue-storefront/core';
```

File: packages/api-client/src/api/updateCartItems/index.ts [hunk 13]
```diff
@@ -1,23 +1,29 @@
...
-import updateCartItems from './updateCartItems';
+import { Context } from '../../types/context';
```
Comment (Frodigo) [inline packages/api-client/src/api/updateCartItems/index.ts:3]:
```suggestion
import type { Context } from '../../types/context';
```

**Review by Frodigo — CHANGES_REQUESTED** [review]

File: docs/.vuepress/config.js [hunk 14]
```diff
@@ -107,14 +109,14 @@ module.exports = {
           ['/api-reference/magento-api.deletecustomeraddress', 'deleteCustomerAddress'],
           ['/api-reference/magento-api.getcustomeraddresses', 'getCustomerAddresses'],
           ['/api-reference/magento-api.updatecustomeraddress', 'updateCustomerAddress'],
-        ]
+        ],
```
Comment (Frodigo) [inline docs/.vuepress/config.js:112]: I can't see new documented API methods here. Could you add them?

Comment (roziscoding) [reply docs/.vuepress/config.js:122]: Done :)

**Review by Frodigo — APPROVED** [review]

**Pull request merged**